src/main/java/org/apache/gossip/manager/PassiveGossipThread.java

```diff
@@ -53,8 +53,11 @@
   private final ObjectMapper MAPPER = new ObjectMapper();
   
   private final GossipCore gossipCore;
+  
+  protected final CountDownLatch gate;
 
-  public PassiveGossipThread(GossipManager gossipManager, GossipCore gossipCore) {
+  public PassiveGossipThread(GossipManager gossipManager, GossipCore gossipCore,
```

Comment:
Passed-in CountDownLatch in PassiveGossipThread and ActiveGossipThread actually is a shared instance, constructed by ServiceFactory.newServices() function, to control the count down latch through CountDownLatch.countDown(). If constructing CountDownLatch within ActiveGossipThread and PassiveGossipThread individually then the GossipManager thread may not pass await() function because CountDownLatch instance in GossipManager will not count down to zero. Or any strategies I can apply to construct CountDownLatch within active and passive threads without sharing the CountDownLatch instance?